src/Orchard.Web/Modules/Orchard.Layouts/Migrations.cs

```diff
@@ -89,5 +103,32 @@ public class Migrations : DataMigrationImpl {
                 .WithSetting("Stereotype", "Widget")
                 .DisplayedAs(widgetDisplayedAs));
         }
+        public int UpdateFrom3(){
+            Dictionary<int, string> idAliasDictionary = new Dictionary<int, string>();
+            var queryParts = _contentManager.Query<QueryPart, QueryPartRecord>().List();
+            foreach(var queryPart in queryParts) {
+                foreach (var layout in queryPart.Layouts) {
+                    layout.Alias = Guid.NewGuid().ToString("n");
+                    idAliasDictionary.Add(layout.Id, layout.Alias);
+                }
+            }
+            var layoutParts = _contentManager.Query<LayoutPart, LayoutPartRecord>().List();
```

Comment:
We should not update the database during the migrations if possible. We should instead use the Update module to execute this manually as it will handle batches and won't pollute the migrations.

Also you could use `Identity` instead of `Alias` to use a more common term that does the same thing for content items.